### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by jdelga**
Update Android Logger dependencies

* Use Toml file (no auto-update for now)
* Update to Java 17
* Update libraries

### CODE REVIEW

File: app/src/main/java/com/telefonica/androidlogger/app/MainActivity.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ class MainActivity : AppCompatActivity() {
 
     private fun setupViews() {
         findViewById<Button>(R.id.button_add_logs).setOnClickListener {
-            (1..NUMBER_OF_LOGS_TO_ADD).forEach {
+            repeat(NUMBER_OF_LOGS_TO_ADD) {
```
Comment (jdelga) [inline app/src/main/java/com/telefonica/androidlogger/app/MainActivity.kt:23]: This was a newer lint issue

**Review by juangardi21 — APPROVED** [review]
LGTM 👍🏽 thanks!

File: gradle.properties [hunk 2]
```diff
@@ -14,12 +14,8 @@ org.gradle.jvmargs=-Xmx6656m -Dfile.encoding=UTF-8
 org.gradle.parallel=true
 
 android.useAndroidX=true
-kapt.incremental.apt=true
```
Comment (pmartinbTEF) [inline gradle.properties:17]: Why is this line deleted?

Comment (jdelga) [reply gradle.properties:17]: Some configuration has been deprecated or removed when updating Kotlin or Gradle. In this particular case, kapt was not even used.

**Review by pmartinbTEF — APPROVED** [review]
Good job!
